### ISSUE TITLE
4.0.0.beta1 unknown attribute transaction_id

### ISSUE THREAD

**Issue by jaredbeck** [airblade/paper_trail#455]
After upgrading to 4.0.0.beta1 my tests fail with:

```
ActiveRecord::UnknownAttributeError: unknown attribute 'transaction_id' for PaperTrail::Version.
```

I see that new migrations have been added to the generators:

```
lib/generators/paper_trail/templates/add_transaction_id_column_to_versions.rb
lib/generators/paper_trail/templates/create_version_associations.rb
```

The only instructions I've found:

> PaperTrail can restore three types of associations: Has-One, Has-Many, and Has-Many-Through. In order to do this, you will need to create a version_associations table, either at installation time with the rails generate paper_trail:install --with-associations option or manually.

I don't need this feature.  Is there a way to turn it off so that I can avoid schema changes?  If not, does "manually" mean that I should copy these migrations to my migrations folder?

Given the schema changes and the eager_load! issue with custom initializers, may I suggest adding an "Upgrading to 4.0" section to the readme?

**Comment by jaredbeck** [comment]
> .. pull down the current master branch and give it a whirl .. without those additional migrations migrated

After `rollback` and switching to master I'm not seeing UnknownAttributeError.  All my tests pass.  Thanks!

> The association piece is supposed to be entirely optional

That's wise, you'll probably get more adoption of the new version.

> I just pushed up a commit (899c15b)

To play "back seat driver" for a sec: you're duplicating the `paper_trail_version_class.column_names.include?('transaction_id')` condition in a few places.  I don't know the first thing about the paper_trail codebase, but maybe you could extract a method?

**Comment by jetabe** [comment]
this is happening to us as well.. i hope you are able to releae soon..

**Comment by jaredbeck** [comment]
> this is happening to us as well.. i hope you are able to releae soon..

For now, you can use master, or stay on 3.0.6, depending on whether you need to reify associations or not.  Ben has been really helpful and quick with my questions, so I'm sure it won't be long until another beta of 4.0 shows up.

**Comment by batter** [comment]
Closed by 899c15b

**Comment by jaredbeck** [comment]
@batter Is it much trouble to cut another beta release with https://github.com/airblade/paper_trail/commit/899c15b24dd235cd1adfbf521b658babba2e871f in it?  Not a big deal, I've been using your master branch.  Thanks again.

**Comment by batter** [comment]
Going to try to cut a new release by end of day today

**Comment by batter** [comment]
Just released `v4.0.0.beta2`

**Comment by jaredbeck** [comment]
Switching from `master` to `4.0.0.beta2`, not seeing `UnknownAttributeError`, thanks for cutting another release!
